tests/averaging/rotation/test_rotation_averaging_base.py

```diff
@@ -22,6 +23,54 @@ def setUp(self):
 
         self.obj = DummyRotationAveraging()
 
+    def test_circle_two_edges(self):
+        """Tests for 4 poses in a circle, with a pose connected to its immediate neighborhood."""
+
+        if isinstance(self.obj, DummyRotationAveraging):
```

Comment:
Can we delete the Dummy class now? We have a fully developed Shonan averaging class as an example, so I don't see its purpose. This would be much tidier without that class.

Reply:
Otherwise, if you write a helper as suggested in my other comment, we could move this check into it too.

Reply:
@johnwlambert  what do you think of this? The reason for having Dummy class can be keeping the option to write custom averaging APIs, either by the GTSFM team or our users.

Reply:
I feel that Shonan is just as good of an example of a sample API as the Dummy example, so I'm with Akshay that I would like to see the Dummy disappear.

I haven't had time to do a full review of this PR yet, I will make a pass on Monday.